core/src/test/java/hudson/FilePathTest.java

```diff
@@ -325,6 +328,67 @@ private FilePath createFilePath(final File base, final String... path) throws IO
         FileUtils.touch(building);
         return new FilePath(building);
     }
+    
+    /**
+     * Performs round-trip archiving for Tar handling methods.
+     * @throws Exception test failure
+     */
+    @Test public void compressTarUntarRoundTrip() throws Exception {
+        checkTarUntarRoundTrip("compressTarUntarRoundTrip_zero", 0);   
+        checkTarUntarRoundTrip("compressTarUntarRoundTrip_small", 100); 
+        checkTarUntarRoundTrip("compressTarUntarRoundTrip_medium", 50000); 
+    }
+            
+    /**
+     * Checks that big files (>8GB) can be archived and then unpacked.
+     * This test is disabled by default due the impact on RAM.
+     * The actual file size limit is 8589934591 bytes.
+     * @throws Exception test failure
+     */
+    @Issue("JENKINS-10629")
+    @Ignore
+    @Test public void archiveBigFile() throws Exception {
+        final long largeFileSize = 9000000000L; // >8589934591 bytes
+        final String filePrefix = "JENKINS-10629";
+        checkTarUntarRoundTrip(filePrefix, largeFileSize);
+    }
+     
+    private void checkTarUntarRoundTrip(String filePrefix, long fileSize) throws Exception {
+        File tempFile = null, tarFile = null;
+        FilePath outDir = null;
+        
+        try {
+            tempFile = File.createTempFile(filePrefix, ".log");
```

Comment:
You can just use `FilePathTest.temp` to allocate temporary files without bothering with a `finally`-block.